.github/workflows/build.yml

```diff
@@ -37,7 +36,7 @@ jobs:
       - uses: docker/build-push-action@v3
         with:
           context: ${{ matrix.image }}
-          push: true
+          push: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
This seemed to me like the situation where we want to do this: only on the `main` branch.

Please let me know if you think this is not the optimal way to do this.

Reply:
I would do `${{ github.event_name != 'pull_request' }}`

Reply:
Out of curiosity, what is the advantage of one over the other here?

Reply:
If we wanted to move to another default branch. Not really much else, just a pattern we've done [before](https://github.com/articulate/docker-articulate-elixir/blob/96db8cc0c85ce06aa7a26068743c1df12cd92a71/.github/workflows/build.yml#L34)

Reply:
Also ref is weird when it comes to pull_requests, which doesn't matter too much in this context

Reply:
Thank you!